### PR TITLE
Added the following two helper functions for modifiers

### PR DESCRIPTION
**Pull request by rdaly525**
strip_modifiers removes all type modifiers within an ADT type
push_modifiers pushes all the type modifiers to the leafs of the ADT

automapper in peak will be using these, but they seemed generic/useful enough to add as helpers in hwtypes

### CODE REVIEW

File: hwtypes/modifiers.py [hunk 1]
```diff
@@ -128,3 +131,46 @@ def wrap_modifier(T, mods):
...
+
+#Takes an ADT type and removes any modifiers on any level of the Tree
+def strip_modifiers(adt_t):
+    #remove modifiers from this level
+    adt_t, _ = unwrap_modifier(adt_t)
+    if (issubclass(adt_t, AbstractBit) or issubclass(adt_t, AbstractBitVector)):
+        return adt_t
+    elif issubclass(adt_t, Enum):
+        return adt_t
+    elif issubclass(adt_t, Sum):
```
Comment (cdonovick) [inline hwtypes/modifiers.py]: Check `issubclass(adt_t, GetitemSyntax)`, for Sum/Tuple, build new type with `adt_t.unbound_t[new_fields]`
Check `issubclass(adt_t, AttrSyntax) and adt.FIELD_T is type` for Product/TaggedUnion and exclude Enum.  Build new type similarly to above.
else return adt_t

File: hwtypes/modifiers.py [hunk 2]
```diff
@@ -128,3 +131,46 @@ def wrap_modifier(T, mods):
...
+        return Product.from_fields(adt_t.__name__, new_fields)
+    elif issubclass(adt_t, Tuple):
+        new_fields = [strip_modifiers(sub_adt_t) for sub_adt_t in adt_t.fields]
+        return Tuple[new_fields]
+    else:
+        raise ValueError("Unreachable")
+
+#Takes an ADT type and pushes all the modifiers from internal nodes to leaf nodes
+def push_modifiers(adt_t, mods=[]):
+    #remove modifiers from this level
```
Comment (cdonovick) [inline hwtypes/modifiers.py:151]: same as above

**Review by cdonovick — CHANGES_REQUESTED** [review]

File: hwtypes/modifiers.py (hunk 1 above)
Comment (rdaly525) [reply hwtypes/modifiers.py]: @cdonovick do you mean issubclass(type(adt_t), AttrSyntax) ?

Comment (cdonovick) [reply hwtypes/modifiers.py]: I meant `isinstance`

**Comment by rdaly525** [comment]
@cdonovick, I consider my current code to be significantly clearer in terms of user intent opposed to your requests. I would recommend that we merge this now, and if you want, you describe precisely how to write conditions in a way you consider to be better style.

**Comment by cdonovick** [comment]
@rdaly525 
The issue is there should be nothing special about, enum, bitvector, bit, etc... arbitrary leaf types should work without this code changing.

**Comment by rdaly525** [comment]
@cdonovick, updated the code to how we agreed. Now depends on #116 being fixed

**Comment by rdaly525** [comment]
Should be fixed by #117 @cdonovick can you approve?

**Review by cdonovick — APPROVED** [review]

**Pull request merged**